page-tracking/src/collections/page_box.rs

```diff
@@ -321,4 +373,58 @@ mod tests {
             )
             .is_ok());
     }
+
+    #[test]
+    fn page_drop() {
+        const TEST_PAGE_COUNT: usize = 3;
+        const TEST_BYTES: usize = TEST_PAGE_COUNT * PageSize::Size4k as usize;
+        let (page_tracker, mut pages) = PageTracker::new_in_test();
+        let assigned_pages =
+            SequentialPages::from_pages(pages.by_ref().take(TEST_PAGE_COUNT).map(|p| {
+                page_tracker
+                    .assign_page_for_internal_state(p, PageOwnerId::host())
+                    .unwrap()
+            }))
+            .unwrap();
+
+        let mut addr = assigned_pages.base();
+        let page_size = assigned_pages.page_size();
+        let page_count = assigned_pages.len();
+        assert_eq!(page_count, TEST_PAGE_COUNT as u64);
+
+        {
+            let _pb = PageBox::new_with([5u8; TEST_BYTES], assigned_pages, page_tracker.clone());
+        }
+
+        for _pi in 0..TEST_PAGE_COUNT {
+            // Should go back to hypervisor-owned after the PageBox was dropped.
+            assert!(page_tracker
+                .get_converted_page::<Page<ConvertedDirty>>(
+                    addr,
+                    PageOwnerId::hypervisor(),
+                    TlbVersion::new()
+                )
+                .is_ok());
+            addr = addr.checked_add_pages_with_size(1, page_size).unwrap();
+        }
+    }
+
+    #[test]
+    #[should_panic(
```

Comment:
nit: I'd just use a plan `should_panic`; testing against strings is kinda brittle and given how simple this test is I'm not sure it's worth guarding against multiple possible panic paths.

Reply:
Sure, makes sense.